metrics/metrics.go

```diff
@@ -16,7 +16,9 @@ var (
 	setupErr     error
 	statsdClient statsd.ClientInterface
 	errorHandler errors.ErrorHandler
-	opts         *options
+	opts         = &options{
+		metricSampleRate: float64(0),
+	}
```

Comment:
I think this should be moved to the same location as where we are setting the `noopClient`. That will make it clearer why we are setting these values. Consider creating a function `resolveNoopOptions`.